src/token.rs

```diff
@@ -24,92 +32,134 @@ impl Default for TokenKind {
     }
 }
 
-/// script of a token (https://docs.rs/whatlang/0.10.0/whatlang/enum.Script.html)
 #[derive(Debug, Clone, Default)]
-pub struct Token<'a> {
+pub struct Token<'o> {
+    /// kind of the Token assigned by the classifier
     pub kind: TokenKind,
-    pub word: Cow<'a, str>,
-    /// index of the first character of the word
-    pub char_index: usize,
-    /// indexes of start and end of the byte slice
+    pub lemma: Cow<'o, str>,
+    /// index of the first and the last character of the original lemma
+    pub char_start: usize,
+    pub char_end: usize,
+    /// index of the first and the last byte of the original lemma
     pub byte_start: usize,
     pub byte_end: usize,
     /// number of bytes used in the normalized string
-    ///  by each char in the original string
+    /// by each char in the original string
```

Comment:
Maybe you can add that the `char_map` must be the same length as the number of `char`s in the `str`?